clients/js/src/index.ts

```diff
@@ -104,18 +104,23 @@ export class Collection {
       this.embeddingFunction = embeddingFunction;
   }
 
-  public async add(
+  private async validate(
```

Comment:
small improvements here:
1. I don't think this needs to be `async` since there is no `async` requirement inside of it
2. should this be a method on the client, or just be a function that can be used inside the client? (it does currently reference `this.embeddingFunction`) 
3. can we improve the naming around `require_embeddings`? or remove it? Both places in this file have it to be `true`. is it ever `false`?

Reply:
Will add a comment `require_embeddings=False` in the case of `update` (which is in main, but not this PR yet)